lib/core/connector_settings.rb

```diff
@@ -21,17 +21,40 @@ class ConnectorSettings
     DEFAULT_REDUCE_WHITESPACE = true
     DEFAULT_RUN_ML_INFERENCE = false
 
+    DEFAULT_PAGE_SIZE = 100
+
     # Error Classes
     class ConnectorNotFoundError < StandardError; end
 
-    def self.fetch(connector_id)
+    def self.fetch_by_id(connector_id)
       es_response = ElasticConnectorActions.get_connector(connector_id)
       globals = ElasticConnectorActions.connectors_meta
 
       raise ConnectorNotFoundError.new("Connector with id=#{connector_id} was not found.") unless es_response[:found]
       new(es_response, globals)
     end
 
+    def self.fetch_native(page_size = DEFAULT_PAGE_SIZE)
+      globals = ElasticConnectorActions.connectors_meta
```

Comment:
Just curious why `globals` and not connectors_meta? It confuses me a bit.

Reply:
+1, I think `connectors_meta` or even `meta` is a much better name.

Reply:
Also, this variable is not global :)